app/build.gradle

```diff
@@ -1,30 +1,20 @@
 import org.apache.tools.ant.filters.EscapeUnicode
 
-buildscript {
-    repositories { jcenter() }
-    dependencies {
-        classpath 'com.github.jengelman.gradle.plugins:shadow:2.0.4'
-    }
+plugins {
+    id 'application'
+    id "com.github.johnrengelman.shadow" version "4.0.3"
 }
 
-apply plugin: 'application'
-
 mainClassName = 'com.oracle.javafx.scenebuilder.app.SceneBuilderApp'
 
-apply plugin: 'com.github.johnrengelman.shadow'
-
 dependencies {
-    compile project(':kit')
+    implementation project(':kit')
 }
 
 run {
     doFirst {
-        jvmArgs = [
-                '--module-path', classpath.asPath,
-                '--add-modules', 'javafx.fxml,javafx.swing,javafx.web,java.logging,java.scripting',
-                '--add-opens=javafx.fxml/javafx.fxml=ALL-UNNAMED'
-        ]
-        if (platform == 'linux') jvmArgs += '-Djdk.gtk.version=2'
+        jvmArgs += ['--add-opens=javafx.fxml/javafx.fxml=ALL-UNNAMED' ]
+        if (javafx.platform == 'linux') jvmArgs += '-Djdk.gtk.version=2'

```

Comment:
I missed this before, but you shoulc check the value of either `javafx.platform.classifier` or `osdetector.os`.